harper-core/src/spell/mod.rs

```diff
@@ -212,31 +188,200 @@ mod tests {
     }
 
     #[test]
-    fn simple1() {
-        assert_edit_dist("kitten", "sitting", 3)
+    fn simple_edit_distance_1() {
+        assert_edit_dist("kitten", "sitting", MAX_EDIT_DIST)
     }
 
     #[test]
-    fn simple2() {
-        assert_edit_dist("saturday", "sunday", 3)
+    fn simple_edit_distance_2() {
+        assert_edit_dist("saturday", "sunday", MAX_EDIT_DIST)
     }
 
     #[test]
     fn produces_no_duplicates() {
-        let results =
-            suggest_correct_spelling_str("punctation", 100, 3, &FullDictionary::curated());
+        let results = suggest_correct_spelling_str(
+            "punctation",
+            RESULT_LIMIT,
+            MAX_EDIT_DIST,
+            &FstDictionary::curated(),
+        );
 
         dbg!(&results, results.iter().unique().collect_vec());
 
         assert_eq!(results.iter().unique().count(), results.len())
     }
 
+    #[test]
+    fn zero_edit_distance() {
+        let source: Vec<_> = "hello".chars().collect();
+        let target: Vec<_> = "hello".chars().collect();
+        assert_eq!(edit_distance(&source, &target), 0);
+    }
+
+    #[test]
+    fn one_edit_distance() {
+        let source: Vec<_> = "hello".chars().collect();
+        let target: Vec<_> = "hellos".chars().collect();
+        assert_eq!(edit_distance(&source, &target), 1);
+        assert_eq!(edit_distance(&target, &source), 1);
+
+        let target: Vec<_> = "hell".chars().collect();
+        assert_eq!(edit_distance(&source, &target), 1);
+        assert_eq!(edit_distance(&target, &source), 1);
+
+        let target: Vec<_> = "hell".chars().collect();
+        assert_eq!(edit_distance(&source, &target), 1);
+        assert_eq!(edit_distance(&target, &source), 1);
+
+        let target: Vec<_> = "hvllo".chars().collect();
+        assert_eq!(edit_distance(&source, &target), 1);
+        assert_eq!(edit_distance(&target, &source), 1);
+
+        let target: Vec<_> = "Hello".chars().collect();
+        assert_eq!(edit_distance(&source, &target), 1);
+        assert_eq!(edit_distance(&target, &source), 1);
+    }
+
+    #[test]
+    fn common_words_first() {
```

Comment:
I'm a little confused. What is this test doing? It feels really complicated and difficult to read.

Reply:
This just checked that the algorithm prioritizes common words by getting both common and uncommon words and then running the algorithm over them.

I agree it is a poorly written test though.

Reply:
Would you mind adding a bit of documentation that makes it clearer? I'm sorry for all my nitpicks, but this should be the last one!